src/kernel/src/fs/file.rs

```diff
@@ -67,13 +68,104 @@ pub trait VFileOps: Debug + Send + Sync + Display {
     fn ioctl(
         &self,
         file: &VFile,
-        com: u64,
+        com: IoctlCom,
         data: &mut [u8],
         cred: &Ucred,
         td: &VThread,
     ) -> Result<(), Box<dyn Errno>>;
 }
 
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct IoctlCom(u32);
+
+impl Display for IoctlCom {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{:x}", self.0)
+    }
+}
+
+impl IoctlCom {
+    pub const IOCPARM_SHIFT: u32 = 13;
+    pub const IOCPARM_MASK: u32 = (1 << Self::IOCPARM_SHIFT) - 1;
+    pub const IOC_VOID: u32 = 0x20000000;
+    pub const IOC_OUT: u32 = 0x40000000;
+    pub const IOC_IN: u32 = 0x80000000;
+    pub const IOC_INOUT: u32 = Self::IOC_IN | Self::IOC_OUT;
+
+    pub const fn try_from_raw(com: u64) -> Result<Self, SysErr> {
```

Comment:
Better to return `Option` and let the caller decide what to do instead.

Reply:
Good idea.